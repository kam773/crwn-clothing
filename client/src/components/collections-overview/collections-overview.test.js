import React from 'react';
import { shallow, mount } from 'enzyme';
import { CollectionsOverview } from './collections-overview.component';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('create shapshot', () => {
  const initialState = {
    collections: [
      {
        routeName: '/mens',
        id: '12',
        title: 'Mens',
        items: [
          { id: '1', imageUrl: 'url', price: 200 },
          { id: '1', imageUrl: 'url', price: 200 },
        ],
      },
      {
        routeName: '/mens',
        id: '12',
        title: 'Mens',
        items: [
          { id: '1', imageUrl: 'url', price: 200 },
          { id: '1', imageUrl: 'url', price: 200 },
        ],
      },
      {
        routeName: '/mens',
        id: '12',
        title: 'Mens',
        items: [
          { id: '1', imageUrl: 'url', price: 200 },
          { id: '1', imageUrl: 'url', price: 200 },
        ],
      },
      {
        routeName: '/mens',
        id: '12',
        title: 'Mens',
        items: [
          { id: '1', imageUrl: 'url', price: 200 },
          { id: '1', imageUrl: 'url', price: 200 },
        ],
      },
    ],
  };

  const mockStore = configureStore();
  let store;
  let container;

  store = mockStore(initialState);
  container = shallow(
    <MemoryRouter>
      <CollectionsOverview {...initialState} />
    </MemoryRouter>,
  ).dive();

  it('captures snapshot of CollectionsOverview', () => {
    const wrapper = renderer.create(container).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
