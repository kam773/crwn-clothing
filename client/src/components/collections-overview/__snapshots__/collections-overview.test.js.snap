// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create shapshot captures snapshot of CollectionsOverview 1`] = `
<div
  className="sc-iwsKbI dRZuUw"
>
  <div
    className="sc-gzVnrw bvbgQA"
  >
    <h1
      className="sc-htoDjs kCgwzc"
      onClick={[Function]}
    >
      MENS
    </h1>
    <div
      className="sc-dnqmqq koSTAP"
      data-test="preview-container"
    >
      <div
        className="sc-bwzfXH cFLGUT"
      >
        <div
          className="sc-bxivhb dbDlmL image"
        />
        <div
          className="sc-ifAKCX kDnhkJ"
        >
          <span
            className="sc-EHOje kkQgTP"
          />
          <span
            className="sc-bZQynM gGKQi"
          >
            200
          </span>
        </div>
        <button
          className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
          onClick={[Function]}
        >
          Add to cart
        </button>
      </div>
      <div
        className="sc-bwzfXH cFLGUT"
      >
        <div
          className="sc-bxivhb dbDlmL image"
        />
        <div
          className="sc-ifAKCX kDnhkJ"
        >
          <span
            className="sc-EHOje kkQgTP"
          />
          <span
            className="sc-bZQynM gGKQi"
          >
            200
          </span>
        </div>
        <button
          className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
          onClick={[Function]}
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw bvbgQA"
  >
    <h1
      className="sc-htoDjs kCgwzc"
      onClick={[Function]}
    >
      MENS
    </h1>
    <div
      className="sc-dnqmqq koSTAP"
      data-test="preview-container"
    >
      <div
        className="sc-bwzfXH cFLGUT"
      >
        <div
          className="sc-bxivhb dbDlmL image"
        />
        <div
          className="sc-ifAKCX kDnhkJ"
        >
          <span
            className="sc-EHOje kkQgTP"
          />
          <span
            className="sc-bZQynM gGKQi"
          >
            200
          </span>
        </div>
        <button
          className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
          onClick={[Function]}
        >
          Add to cart
        </button>
      </div>
      <div
        className="sc-bwzfXH cFLGUT"
      >
        <div
          className="sc-bxivhb dbDlmL image"
        />
        <div
          className="sc-ifAKCX kDnhkJ"
        >
          <span
            className="sc-EHOje kkQgTP"
          />
          <span
            className="sc-bZQynM gGKQi"
          >
            200
          </span>
        </div>
        <button
          className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
          onClick={[Function]}
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw bvbgQA"
  >
    <h1
      className="sc-htoDjs kCgwzc"
      onClick={[Function]}
    >
      MENS
    </h1>
    <div
      className="sc-dnqmqq koSTAP"
      data-test="preview-container"
    >
      <div
        className="sc-bwzfXH cFLGUT"
      >
        <div
          className="sc-bxivhb dbDlmL image"
        />
        <div
          className="sc-ifAKCX kDnhkJ"
        >
          <span
            className="sc-EHOje kkQgTP"
          />
          <span
            className="sc-bZQynM gGKQi"
          >
            200
          </span>
        </div>
        <button
          className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
          onClick={[Function]}
        >
          Add to cart
        </button>
      </div>
      <div
        className="sc-bwzfXH cFLGUT"
      >
        <div
          className="sc-bxivhb dbDlmL image"
        />
        <div
          className="sc-ifAKCX kDnhkJ"
        >
          <span
            className="sc-EHOje kkQgTP"
          />
          <span
            className="sc-bZQynM gGKQi"
          >
            200
          </span>
        </div>
        <button
          className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
          onClick={[Function]}
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw bvbgQA"
  >
    <h1
      className="sc-htoDjs kCgwzc"
      onClick={[Function]}
    >
      MENS
    </h1>
    <div
      className="sc-dnqmqq koSTAP"
      data-test="preview-container"
    >
      <div
        className="sc-bwzfXH cFLGUT"
      >
        <div
          className="sc-bxivhb dbDlmL image"
        />
        <div
          className="sc-ifAKCX kDnhkJ"
        >
          <span
            className="sc-EHOje kkQgTP"
          />
          <span
            className="sc-bZQynM gGKQi"
          >
            200
          </span>
        </div>
        <button
          className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
          onClick={[Function]}
        >
          Add to cart
        </button>
      </div>
      <div
        className="sc-bwzfXH cFLGUT"
      >
        <div
          className="sc-bxivhb dbDlmL image"
        />
        <div
          className="sc-ifAKCX kDnhkJ"
        >
          <span
            className="sc-EHOje kkQgTP"
          />
          <span
            className="sc-bZQynM gGKQi"
          >
            200
          </span>
        </div>
        <button
          className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
          onClick={[Function]}
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
